### PR TITLE
Auto Batch: if plugin is disabled during cmake

### PR DESCRIPTION
**Pull request by jane-intel**
### Details:
 - *Auto Batch: if plugin is disabled during cmake*

### Tickets:
 - *88449*

### CODE REVIEW

**Review by ilyachur — APPROVED** [review]

File: src/inference/src/ie_core.cpp [hunk 1]
```diff
@@ -651,6 +651,12 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
             // as the result is being checked by the user
             strictly_check_dims = false;
         } else {
+            // check if Auto-Batch plugin registered
+            try {
+                GetCPPPluginByName("BATCH");
+            } catch (...) {
```
Comment (ilya-lavrenov) [inline src/inference/src/ie_core.cpp:657]: maybe we need to catch only std::runtime_error https://github.com/openvinotoolkit/openvino/blob/master/src/common/util/src/os/lin/lin_shared_object_loader.cpp#L33? other exceptions can be re-thrown to users

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by songbell** [comment]
@jane-intel  Hi, when will auto  batch plugin be disabled? for some engineering builds or some target releases?

**Comment by jane-intel** [comment]
In case user build ov from source and uses -DENABLE_AUTO_BATCH=OFF for some reason

**Comment by ilya-lavrenov** [comment]
> @jane-intel Hi, when will auto batch plugin be disabled? for some engineering builds or some target releases?

or if user explicitly has not installed / removed auto batch plugin

**Pull request merged**